mayastor/src/bdev/nexus/mod.rs

```diff
@@ -125,6 +132,38 @@ pub fn instances() -> &'static mut Vec<Box<Nexus>> {
     nexus_module::NexusModule::get_instances()
 }
 
+pub async fn nexus_fault_child(child_name: String) {
+    let cn = &child_name;
+    for nexus in instances() {
+        if nexus
+            .children
+            .iter()
+            .filter(|c| c.state() == ChildState::Open)
+            .filter(|c| {
+                if let Ok(child) = c.get_device().as_ref() {
+                    child.device_name() == *cn
+                } else {
+                    false
+                }
+            })
+            .any(|c| {
+                Ok(ChildState::Open)
+                    == c.state.compare_exchange(
+                        ChildState::Open,
+                        ChildState::Faulted(Reason::IoError),
+                    )
+            })
+        {
+            info!("{} retiring {}", nexus.name, child_name);
+            // Error indicates it is already paused and another thread is
+            // processing the fault
+            let _ = nexus.child_retire(child_name).await;
```

Comment:
Should we output a message in the event of an error? It might help with future debugging.

Reply:
Sure, I'll add that.